demos/imagenet_classification_demo/grid_mat.hpp

```diff
@@ -0,0 +1,170 @@
+        testMessage = "Testing, please wait...";
+        testMessageSize = cv::getTextSize(testMessage, fontType, fontScale, thickness, &baseline);
+    }
+
+    void textUpdate(double avgFPS, double avgLatency, double accuracy,
+                    bool isFpsTest, bool showAccuracy,
+                    Presenter& presenter) {
+        rectangle(outImg,
+                  {0, 0}, {outImg.cols, presenter.graphSize.height},
+                  cv::Scalar(0, 0, 0), cv::FILLED);
+
+        presenter.drawGraphs(outImg);
+        
+        cv::Scalar textColor = cv::Scalar(255, 255, 255);
+        int textPadding = 10;
+
+        cv::putText(outImg,
+                    cv::format("FPS: %0.01f", avgFPS),
+                    cv::Point(textPadding, textSize.height + textPadding),
+                    fontType, fontScale, textColor, thickness);
+        cv::putText(outImg,
+                    cv::format("Latency: %dms", static_cast<int>(avgLatency * 1000)),
+                    cv::Point(textPadding, (textSize.height + textPadding) * 2),
+                    fontType, fontScale, textColor, thickness);
+        
+        if (showAccuracy) {
+            cv::putText(outImg,
+                        cv::format("Accuracy (top %d): %.3f", FLAGS_nt, accuracy),
+                        cv::Point(outImg.cols - accuracyMessageSize.width - textPadding, textSize.height + textPadding),
+                        fontType, fontScale, textColor, thickness);
+        }
+
+        if (isFpsTest) {
+            cv::putText(outImg,
+                        testMessage,
+                        cv::Point(outImg.cols - testMessageSize.width - textPadding,
+                                  (textSize.height + textPadding) * 2),
+                        fontType, fontScale, cv::Scalar(50, 50, 255), thickness);
+        }
+    }
+
+    void updateMat(ImageInfoList& imageInfos) {
+        ImageInfoList imagesToDraw;
+        if (!imageInfos.empty()) {
+            imagesToDraw.splice(imagesToDraw.end(), imageInfos);
+        }
+
+        if (!prevImgs.empty()) {
+            int prevSourceId = currSourceId - FLAGS_b;
+            if (prevSourceId < 0) {
+                prevSourceId += points.size();
+            }
+            for (size_t i = 0; i < prevImgs.size(); i++, prevSourceId++) {
+                if (size_t(prevSourceId) >= points.size()) {
+                    prevSourceId -= points.size();
+                }
+
+                prevImgs.front().copyTo(outImg(cv::Rect(points[prevSourceId], cellSize)));
+                prevImgs.pop();
+            }   
+        }
+
+        while (!imagesToDraw.empty()) {
+            cv::Mat frame = std::get<0>(imagesToDraw.front());
+            std::string predictedLabel = std::get<1>(imagesToDraw.front());
+            PredictionResult predictionResult = std::get<2>(imagesToDraw.front());
+            imagesToDraw.pop_front();
```

Comment:
If this is removed, it will be possible to accept `imageInfos` by const ref and iterate over it without moving elements to another container.